### PR TITLE
Adjust calculation of child frame position to work on more backends

### PR DESCRIPTION
**Pull request by fejfighter**
* src/pgtkterm.c (x_set_offset): rework position calculations

### CODE REVIEW

**Comment by fejfighter** [comment]
fixes #46

**Comment by fejfighter** [comment]
I might have a better method for this, as it stands it's works well on X and wayland, but not broadway.
This would imply that I am missing something and it could affect other future ports

**Comment by fejfighter** [comment]
I think the broadway issue is related to scaling of some sort.

the "better method" involved `gtk_widget_translate_coordinates ` which appeared to do nothing from all examples found

File: src/pgtkterm.c [hunk 1]
```diff
@@ -300,16 +300,26 @@ x_set_offset (struct frame *f, int xoff, int yoff, int change_gravity)
...
   int surface_pos_x = 0;
   int surface_pos_y = 0;
 
   if (parent)
     {
       /* determing the "height" of the titlebar, by finding the
 	 location of the "emacsfixed" widget on the surface/window */
+      GtkAllocation a = { 0 };
       GtkWidget *w = FRAME_GTK_WIDGET (parent);
+      GdkWindow *gw = gtk_widget_get_window(w);
```
Comment (masm11) [inline src/pgtkterm.c]: could you put a space after the function name?

File: src/pgtkterm.c [hunk 2]
```diff
@@ -300,16 +300,26 @@ x_set_offset (struct frame *f, int xoff, int yoff, int change_gravity)
...
+      GdkRectangle rect = {0};
+
       gtk_widget_get_allocation (w, &a);
+
+      gdk_window_get_frame_extents(gw, &rect);
```
Comment (masm11) [inline src/pgtkterm.c]: could you put a space after the function name?

**Review by masm11 — CHANGES_REQUESTED** [review]
please fix coding style.

**Comment by fejfighter** [comment]
@masm11 I'll add fixes for #48 and #49 to this pull request

**Comment by fejfighter** [comment]
superseded by #51

**Pull request closed without merging**